projects/packages/masterbar/src/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -0,0 +1,536 @@
+	 *
+	 * @return int
+	 */
+	public function get_current_user_blog_count() {
+		if ( function_exists( '\get_blog_count_for_user' ) ) {
+			return \get_blog_count_for_user( get_current_user_id() );
+		}
+
+		$blogs = get_blogs_of_user( get_current_user_id() );
+		return is_countable( $blogs ) ? count( $blogs ) : 0;
+	}
+
+	/**
+	 * Adds the site switcher link if user has more than one site.
+	 */
+	public function add_browse_sites_link() {
+		if ( $this->get_current_user_blog_count() < 2 ) {
+			return;
+		}
+
+		// Add the menu item.
+		// @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. TODO add link with Trac issue.
+		add_menu_page( __( 'site-switcher', 'jetpack-masterbar' ), __( 'Browse sites', 'jetpack-masterbar' ), 'read', 'https://wordpress.com/sites', null, 'dashicons-arrow-left-alt2', 0 );
+		add_filter( 'add_menu_classes', array( $this, 'set_browse_sites_link_class' ) );
+	}
+
+	/**
+	 * Adds a custom element class for Site Switcher menu item.
+	 *
+	 * @param array $menu Associative array of administration menu items.
+	 * @return array
+	 */
+	public function set_browse_sites_link_class( array $menu ) {
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'site-switcher' !== $menu_item[3] ) {
+				continue;
+			}
+
+			$menu[ $key ][4] = add_cssclass( 'site-switcher', $menu_item[4] );
+			break;
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Adds a link to the menu to create a new site.
+	 */
+	public function add_new_site_link() {
+		if ( $this->get_current_user_blog_count() > 1 ) {
+			return;
+		}
+
+		$this->add_admin_menu_separator();
+		// @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. TODO add link with Trac issue.
+		add_menu_page( __( 'Add New Site', 'jetpack-masterbar' ), __( 'Add New Site', 'jetpack-masterbar' ), 'read', 'https://wordpress.com/start?ref=calypso-sidebar', null, 'dashicons-plus-alt' );
+	}
+
+	/**
+	 * Adds site card component.
+	 */
+	public function add_site_card_menu() {
+		$default         = plugins_url( 'globe-icon.svg', __FILE__ );
+		$icon            = get_site_icon_url( 32, $default );
+		$blog_name       = get_option( 'blogname' ) !== '' ? get_option( 'blogname' ) : $this->domain;
+		$status          = new Status();
+		$is_private_site = $status->is_private_site();
+		$is_coming_soon  = $status->is_coming_soon();
+
+		if ( $default === $icon && blavatar_exists( $this->domain ) ) {
+			$icon = blavatar_url( $this->domain, 'img', 32 );
+		}
+
+		$badge = '';
+		if ( $is_private_site || $is_coming_soon ) {
+			$badge .= sprintf(
+				'<span class="site__badge site__badge-private">%s</span>',
+				$is_coming_soon ? esc_html__( 'Coming Soon', 'jetpack-masterbar' ) : esc_html__( 'Private', 'jetpack-masterbar' )
+			);
+		}
+
+		// @todo is_simple_site_redirect accepts the blog id. How is this even working with the domain?
```

Comment:
Since we are on it, I also had to add a phan suppresion linked to the usage of `is_simple_site_redirect` on WPCOM.
@mmtr Maybe you could confirm this is a bug too so I can provide a fix?

Reply:
It looks like it works because the phpdocs in wpcom are incomplete.

* `is_simple_site_redirect()` documents `int` for its `$blog_id` parameter, which is what Phan complains about here. (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sqbznvaf.cuc%3Se%3Q54noq23q%232095-og).
* All `is_simple_site_redirect()` does with it is pass it to `has_blog_sticker()`. (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sqbznvaf.cuc%3Se%3Q54noq23q%232099-og)
* `has_blog_sticker()` says `int|null`. (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qoybt%2Qfgvpxref.cuc%3Se%3Q3n0839qr%23177-og)
* The first thing `has_blog_sticker()` does is run it through `wpcom_get_blog_id()`. (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qoybt%2Qfgvpxref.cuc%3Se%3Q3n0839qr%23182-og)
* `wpcom_get_blog_id()` takes `null|string|int`. When passed a string, it treats it as a domain name or URL and finds the blog ID corresponding to the domain.  (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qoybt%2Qfgvpxref.cuc%3Se%3Q3n0839qr%2311-og)

Looks like both of those incomplete phpdocs are my fault, I added them when setting up the stubs in D145932 and relied a little too much on the parameter names being "blog_id" and thinking they were the same as other "blog_id" parameters that really are int-only. We should fix those.

Reply:
Thanks Brad and oof as the documentation on WPCOM is really confusing! No action needed here then apart from removing the todo. I'll do that in the next PR touching the `jetpack-masterbar` package

Reply:
Fixing the phpdocs in D153317-code

Reply:
I went ahead and took care of the comment in #38037, since I had to touch the package anyway to update the baseline.